examples/platforms/emsk/flash.c

```diff
@@ -0,0 +1,175 @@
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ *   This file implements the OpenThread platform abstraction for flash.
+ *
+ */
+
+#include <openthread-config.h>
+#include "openthread/platform/alarm.h"
+#include <utils/flash.h>
+#include <common/code_utils.hpp>
+#include "platform-emsk.h"
+
+/**
+ * EMSK HAS 128 Mbit (16 MB) SPI flash memory (Winbond W25Q128BV).
+ * The Winbond W25Q128BV array is organized into 65536 programmable pages of
+ * 256-byte each. Up to 256 bytes can be programmed at a time. Pages can be
+ * erased in groups of 16 (4KB sector erase).
+ */
+
+/**
+ * EMSK SPIflash memory map with using embARC
+ * FPGA image: from 0x0000 0000
+ * Secondary bootloader: from 0x0078 0000
+ * Available for OpenThread: from 0x00FF D000 to 0x00FF EFFF
+ */
+#define OPENTHREAD_FLASH_BASE       0x00ffd000
+#define OPENTHREAD_FLASH_SIZE       0x00002000
+
+/**
+ * Define SETTINGS_CONFIG_BASE_ADDRESS,
+ *    SETTINGS_CONFIG_PAGE_SIZE, and
+ *    SETTINGS_CONFIG_PAGE_NUM for EMSK
+ */
+#ifdef SETTINGS_CONFIG_BASE_ADDRESS
+#undef SETTINGS_CONFIG_BASE_ADDRESS
+#endif // SETTINGS_CONFIG_BASE_ADDRESS
+#define SETTINGS_CONFIG_BASE_ADDRESS    0x00ffe000
+
+#ifdef SETTINGS_CONFIG_PAGE_SIZE
+#undef SETTINGS_CONFIG_PAGE_SIZE
+#endif // SETTINGS_CONFIG_PAGE_SIZE
+#define SETTINGS_CONFIG_PAGE_SIZE   FLASH_SECTOR_SIZE
+
+#ifdef SETTINGS_CONFIG_PAGE_NUM
+#undef SETTINGS_CONFIG_PAGE_NUM
+#endif // SETTINGS_CONFIG_PAGE_NUM
+#define SETTINGS_CONFIG_PAGE_NUM    1
+
+ThreadError utilsFlashInit(void)
+{
+    flash_init();
+    return kThreadError_None;
+}
+
+uint32_t utilsFlashGetSize(void)
+{
+    return (uint32_t)FLASH_SECTOR_SIZE;
+}
+
+ThreadError utilsFlashErasePage(uint32_t aAddress)
+{
+    ThreadError error = kThreadError_None;
+    int32_t status;
+
+    VerifyOrExit((aAddress >= OPENTHREAD_FLASH_BASE) && (aAddress < (OPENTHREAD_FLASH_BASE + OPENTHREAD_FLASH_SIZE - 1)),
+            error = kThreadError_InvalidArgs);
+
+    /* Use 2 sectors in the implementation, cannot erase the address over the boundry */
+    status = flash_erase(aAddress, FLASH_SECTOR_SIZE);
+
+    VerifyOrExit(!(status == -1), error = kThreadError_Failed);
+
+exit:
+    return error;
+}
+
+ThreadError utilsFlashStatusWait(uint32_t aTimeout)
+{
+    ThreadError error = kThreadError_None;
+    uint32_t start = otPlatAlarmGetNow();
+    uint32_t busy = 1;
```

Comment:
Seeing as only the bottom bit of `busy` is used, and its only ever used in conditionals, it should definitely be a bool.  No only does bool make the purpose of `busy` clearer, but it is also more efficient since bool is generally implemented as a `uint8_t`.